src/jira-ticket-check.ps1

```diff
@@ -0,0 +1,75 @@
+param (
+    [string]$jiraDomain,
+    [string]$projectsToFilterTicketsBy,
+    [string]$uniqueProposedBuildIdentifier,
+    [switch]$skipQaApproval = $false,
+    [switch]$deploymentTaskCanBeSubtask = $false
+)
+
+$ProjectsFilter = @($projectsToFilterTicketsBy -split "," | ForEach-Object { " OR project=`"$($_.Trim())`"" }) -join ""
+
+$DeploymentTask = $null
+$DeploymentTaskKey = $null
+
+$Uri = "https://$jiraDomain/rest/api/2/search?jql=(issuetype=`"Deployment Task`" OR issuetype=`"Deployment Sub-Task`"$ProjectsFilter) AND cf[11902]~`"$uniqueProposedBuildIdentifier`""
+
+Write-Output "Generated url to query jira with: $Uri"
+
+$JiraTickets = Invoke-RestMethod -Method Get -Uri $Uri
+
+# Prefer a Deployment Task if there are multiple issues returned
+if ($JiraTickets.total -gt 1) {
+    $DeploymentTask = $JiraTickets.issues | Where-Object { $_.fields.issuetype.name -eq "Deployment Task" }
+}
+
+# If no issues are found, select the first one
+if ($null -eq $DeploymentTask) {
+    $DeploymentTask = $JiraTickets.issues[0]
+}
+
+if ($null -eq $DeploymentTask) {
+    throw "No Deployment task found"
+}
+
+# If multiple issues are found, select the first one
+if ($DeploymentTask.Count -gt 1) {
+    $DeploymentTask = $DeploymentTask[0]
+}
+
+$DeploymentTaskKey = $DeploymentTask.key
+Write-Output "Found the matching Deployment Task $DeploymentTaskKey"
+
+# Technical Approval
+$TechnicalApprovalField = $DeploymentTask.fields.customfield_16323
```

Comment:
Would this customfield work for all of our Jira projects or just the funding ones?

Reply:
I'm 99% sure the answer is yes. I queried all of our custom fields and that was the only one that looked related to technical approval. There is a 1% chance there is another custom field with a weird name that I missed, but if my refactor goes well it should handle that scenario.

Reply:
We're now checking whatever fields are passed in, so if the 1% chance happens then we're still able to use this action!